optimum/intel/openvino/modeling_base.py

```diff
@@ -250,6 +250,13 @@ def fix_op_names_duplicates(model: openvino.runtime.Model):
 
             from optimum.intel.openvino.quantization import _weight_only_quantization
 
+            if not isinstance(quantization_config, dict) and not isinstance(
+                quantization_config, OVWeightQuantizationConfig
+            ):
+                raise RuntimeError(
+                    "Expected quantization_config to be a dictionary or OVWeightQuantizationConfig object."
+                )
```

Comment:
(can be updated but not very important)
```suggestion
                raise TypeError(
                    f"Expected `quantization_config` to be either a dictionary or OVWeightQuantizationConfig object, got {type(quantization_config)}."
                )
```

Reply:
Done